src/Tribe/Plugin_Dependencies.php

```diff
@@ -0,0 +1,183 @@
+<?php defined( 'WPINC' ) or die;
+
+if (! class_exists( 'Tribe__Plugin_Dependencies' ) ) {
+	/**
+	 * Tracks which tribe plugins are currently activated
+	 */
+	class Tribe__Plugin_Dependencies {
+
+		/**
+		 * An multidimensional array of active tribe plugins in the following format
+		 *
+		 * array(
+		 *  'class'   => 'main class name',
+		 *  'version' => 'version num',
+		 *  'path'    => 'Path to the main plugin/bootstrap file'
+		 * )
+		 */
+		protected $active_plugins = array();
+
+		protected static $instance;
+
+		/**
+		 * Static Singleton Factory Method
+		 *
+		 * @return Tribe__Plugin_Dependencies
+		 */
+		public static function instance() {
+			if ( ! isset( self::$instance ) ) {
+				$className      = __CLASS__;
+				self::$instance = new $className;
+			}
+
+			return self::$instance;
+		}
+
+		public function __construct() {
+			$this->add_legacy_plugins();
+		}
+
+
+		/**
+		 * Registers older plugins that did not use this class
+		 */
+		private function add_legacy_plugins () {
+			// Version 4.2 and under of the plugins do not register themselves here, so we'll register them
+			// @TODO Consider removing this in 5.0
+
+			$tribe_plugins = new Tribe__Plugins_List();
+
+			foreach( $tribe_plugins->get_list() as $plugin ) {
```

Comment:
Thanks for the feedback human linter. It made me reread our coding guidelines, which have been updated since the last time I read them. I'll get on these.